src/gc/gc.d

```diff
@@ -1671,6 +1679,43 @@ struct Gcx
         return pool.npages * PAGESIZE;
     }
 
+    /**
+     * Update the thresholds for when to collect the next time
+     */
+    void updateCollectThresholds() nothrow
+    {
+        static float max(float a, float b) nothrow
+        {
+            return a >= b ? a : b;
+        }
+
+        static size_t roundUp(float val, size_t multiple) nothrow
+        {
+            return cast(size_t)(val + multiple - 1 + 0.5) / multiple * multiple;
+        }
+
+        static float zeroDenormal(float val) nothrow
```

Comment:
I suspect I'm repeating myself, but IMO dealing with denormals gratuitiously complicates things, we are not dealing with small numbers (only if the number of used pages is 0 for hundreds of collections).